NitroxTest/Model/OptionalTest.cs

```diff
@@ -78,6 +78,14 @@ public void OptionalValueTypeOrElseValidValue()
             Assert.AreEqual(1, op.OrElse(2));
         }
 
+        [TestMethod]
+        public void OptionalSetValue()
+        {
+            Optional<int> op = Optional<int>.Of(0);
+            op = null;
+            Assert.IsTrue(op.IsEmpty());
```

Comment:
This behavior seems like a bug.  Optional should never be null - if it is, a developer should get a null pointer exception so they know about it instead of silently failing.  

With that being said, that  change likely loses the benefit for having OptionalExtentions.  We may want to just put that code in the optional class unless we have other reasons for keeping it.

Reply:
We could still throw a detailed NPE from the extension method?

Reply:
We could do that, but what is the purpose of the extension methods?

Reply:
In that case it becomes as useless as embedding the function directly in the Optional class, except for a more obvious error message. One might accidentally see an NPE on `op.xyz()` as the value not being set, when `op` itself is in fact `null`.

I haven't settled on making `Optional` a value-type that is default-null though. Right now it seems like we're being explicit on top of being explicit: both when setting _and_ getting a possibly-optional value.

Reply:
The purpose of the extension methods was for when someone sets the Optional to null when assigning a value to it say

```csharp
Optional<randomType> opt = new Optional<randomType(randomTypeValue);
if (opt.IsPresent())
{
    Console.WriteLine("Optional has a value! {0}", opt.Value);
}

opt = null;
if (opt.IsEmpty())
{
    Console.WriteLine("Optional is Empty!");
}

opt = newRandomTypeValue;

if (opt.IsPresent())
{
    Console.WriteLine("Optional Is Present! {0}", opt.Value);
}
```